go/badges/badgestate.go

```diff
@@ -299,12 +310,12 @@ func (b *BadgeState) UpdateWithGregor(ctx context.Context, gstate gregor.State)
 		case "team.delete":
 			var body []teamDeletedBody
 			if err := json.Unmarshal(item.Body().Bytes(), &body); err != nil {
-				b.G().Log.CDebugf(ctx, "BadgeState unmarshal error for team.delete item: %v", err)
+				b.log.CDebugf(ctx, "BadgeState unmarshal error for team.delete item: %v", err)
 				continue
 			}
 
 			msgID := item.Metadata().MsgID().(gregor1.MsgID)
-			username := b.G().Env.GetUsername().String()
+			username := b.env.GetUsername().String()
```

Comment:
should you check `env != nil` here? what will the server pass in?

Reply:
it will be panic if constructed `NewBadgeStateForServer`, but server shouldn't be calling this function right? Should I make it return an error instead?

Reply:
the username check is just so the UI doesn't show a badge for the team you just deleted, i'd just set `username = ""` if `env == nil`

Reply:
🤦‍♂️ missed `NewBadgeStateForServer` but yeah think it's safer to explicitly check